src/app/components/coupon-profile/coupon-profile.component.ts

```diff
@@ -27,12 +30,77 @@ export class CouponProfileComponent implements OnInit {
   ngOnInit(): void {
     this.activatedRoute.params.subscribe((params) => {
       this.generalService.getCoupon(params.id).subscribe((value: Coupon) => {
+        if ((new Date(value.startDate).valueOf()) > new Date().valueOf()) {
+          this.router.navigate([this.authentication.getUrl]);
+          return;
+        }
         this.coupon = value;
+        console.log(this.coupon);
+        this.setButtons();
       });
     });
   }
 
-  private get user(): User {
+
+  setButtons(): void {
+    if (!this.user) {
+      this.purchasable = true;
+    } else {
+      switch (this.user.clientType) {
+        case ClientType.COMPANY:
+          this.couponBelongToTheCompany();
+          break;
+        case ClientType.CUSTOMER:
+          this.couponBelongToTheCustomer();
+          break;
+      }
+    }
+  }
+
+  couponBelongToTheCompany(): void {
+    this.companyService.getDetails().subscribe((value: Company) => {
+      this.editable = this.coupon.companyName === value.name;
```

Comment:
regarding this line, you PUSH everything after the first "="  into this.editable, but is the === statement a condition?
if Its not a condition I don't understand that line.